addons/l10n_in/views/account_invoice_views.xml

```diff
@@ -7,6 +7,8 @@
         <field name="arch" type="xml">
             <xpath expr="//field[@name='ref']" position="after">
                 <field name="country_code" invisible="1"/>
+                <field name="l10n_in_state_id"
+                    attrs="{'invisible': ['|', ('country_code', '!=', 'IN'), ('move_type', '=', 'entry')], 'required': [('country_code', '=', 'IN'), ('move_type', '!=', 'entry')], 'readonly': [('state', '!=', 'draft')]}"/>
```

Comment:
Why was not this in the view already?

Reply:
because before we decide to store value from the backend and propose is that if anyone changes the value in a partner then it's not affect posted invoice.

Reply:
Yes, when we introduce this but now we can view it in UI so users can select different values.